src/utils/config.ts

```diff
@@ -0,0 +1,167 @@
+import assert from "assert";
+import { Signer } from "./ethers";
+import * as constants from "./constants";
+import {
+  HubPool,
+  HubPool__factory,
+  SpokePool,
+  SpokePool__factory,
+} from "@across-protocol/contracts-v2";
+import filter from "lodash/filter";
+
+export type Token = constants.TokenInfo & {
+  l1TokenAddress: string;
+  address: string;
+  isNative: boolean;
+};
+export type TokenList = Token[];
+
+export class ConfigClient {
+  public readonly spokeAddresses: Record<number, string> = {};
+  public readonly spokeChains: Set<number> = new Set();
+  constructor(private config: constants.RouteConfig) {
+    this.init();
+  }
+  private init(): void {
+    this.config.routes.forEach((route) => {
+      this.spokeAddresses[route.fromChain] = route.fromSpokeAddress;
+      this.spokeChains.add(route.fromChain);
+      this.spokeChains.add(route.toChain);
+    });
+  }
+  getRoutes(): constants.Routes {
+    return this.config.routes;
+  }
+  getSpokePoolAddress(chainId: constants.ChainId): string {
+    const address = this.spokeAddresses[chainId];
+    assert(address, "Spoke pool not supported on chain: " + chainId);
+    return address;
+  }
+  getSpokePool(chainId: constants.ChainId, signer?: Signer): SpokePool {
+    const address = this.getSpokePoolAddress(chainId);
+    const provider = signer ?? constants.getProvider(chainId);
+    return SpokePool__factory.connect(address, provider);
+  }
+  getHubPoolChainId(): constants.ChainId {
+    return this.config.hubPoolChain;
+  }
+  getHubPoolAddress(): string {
+    return this.config.hubPoolAddress;
+  }
+  getL1TokenAddressBySymbol(symbol: string) {
+    // all routes have an l1Token address, so just find the first symbol that matches
+    const route = this.config.routes.find((x) => x.fromTokenSymbol);
+    assert(route, `Unsupported l1 address lookup by symbol: ${symbol}`);
+    return route.l1TokenAddress;
+  }
+  getHubPool(signer?: Signer): HubPool {
+    const address = this.getHubPoolAddress();
+    const provider = signer ?? constants.getProvider(this.getHubPoolChainId());
+    return HubPool__factory.connect(address, provider);
+  }
+  filterRoutes(query: Partial<constants.Route>): constants.Routes {
+    return filter(this.config.routes, query);
+  }
+  // this maintains order specified in the constants file in the chainInfoList
+  getSpokeChains(): constants.ChainInfoList {
+    const result: constants.ChainInfoList = [];
+    constants.chainInfoList.forEach((chain) => {
+      if (this.spokeChains.has(chain.chainId)) {
+        result.push(chain);
+      }
+    });
+    return result;
+  }
+  getSpokeChainIds(): constants.ChainId[] {
+    return this.getSpokeChains()
+      .map((chain) => chain.chainId)
+      .filter(constants.isSupportedChainId);
+  }
+  isSupportedChainId(chainId: number): boolean {
+    if (!constants.isSupportedChainId(chainId)) return false;
+    return this.spokeChains.has(chainId);
```

Comment:
nit: a little simpler
```suggestion
    return constants.isSupportedChainId(chainId) && this.spokeChains.has(chainId);
```